src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadata.php

```diff
@@ -114,6 +125,39 @@ public function getDefaultValue()
      */
     public function getAttribute(): ?ArgumentInterface
     {
-        return $this->attribute;
+        trigger_deprecation('symfony/http-kernel', '5.3', 'Method "%s()" is deprecated, use "getAttributes()" instead.', __METHOD__);
+
+        if (!$this->attributes) {
+            return null;
+        }
+
+        return $this->attributes[0] instanceof ArgumentInterface ? $this->attributes[0] : null;
+    }
+
+    /**
+     * @return object[]
+     */
+    public function getAttributes(string $name = null, int $flags = 0): array
+    {
+        if (!$name) {
+            return $this->attributes;
+        }
+
+        $attributes = [];
+        if ($flags & self::IS_INSTANCEOF) {
```

Comment:
Is the flag really needed ? It make sens in the reflection API because doing the instanceof version requires loading the class while the other doesn't but in our cas we already have an instance of it so it doesn't bring much to the table to allow the non-instanceof version.

Reply:
Sure it does. This flag is not here for the reason you mention, see https://3v4l.org/gDqsp.
It's here because one can decide that they do understand the meaning of an exact "Foo" attribute, while they don't want to handle child classes of this Foo attribute.

Reply:
Except I don't see the use case for accepting only the exact attribute and not its child classes (which could be considered a break of LSP).

To me it feels like the instanceof version is the most common use case and should at least be the default if not the only supported case.

Reply:
I think the opposite: attributes are not type-abstractions, they are value objects. But this is going way out of topic and this would deserve way more explanations :)

Reply:
Also I think it's good to have the same flags for scalability, the code can easily handle more flags if php adds more in the future.